test/acceptance/features/steps/steps.py

```diff
@@ -502,8 +545,35 @@ def validate_secret_empty(context, secret_name):
         pass
 
 
+@then(u'The application got redeployed {count} times so far')
+def check_generation(context, count):
+    context.latest_application_generation = context.application.get_generation()
+    assert context.latest_application_generation - context.original_application_generation == int(count), "Unexpected number of application redeployments"
+
+
+@then(u'The application does not get redeployed again with {time} minutes')
+def check_no_redeployment(context, time):
+    try:
+        polling2.poll(lambda: context.application.get_generation() > context.latest_application_generation, step=5, timeout=int(time)*60)
+        assert False, "Application has redeployed again unexpectedly"
+    except polling2.TimeoutException:
+        pass
+
+
 @given(u'"{app_name}" is deployed from image "{image_ref}"')
 def create_deployment(context, app_name, image_ref):
     app = App(app_name, context.namespace.name, image_ref)
     if not app.is_running():
         assert app.install() is True, "Failed to create deployment."
+
+
+@then(u'Secret of Service Binding "{sb_name}" should be updated')
+def validate_secret_name(context, sb_name):
+    openshift = Openshift()
+    time.sleep(10)
+    json_path = "{.status.secret}"
+    output = openshift.get_resource_info_by_jsonpath("servicebindings", sb_name, context.namespace.name, json_path)
+    if output == context.sb_secret_name:
+        assert False, "Secret is not updated"
+    else:
+        assert True
```

Comment:
we should perform pooling and wait until the condition is reached or timeout occurs. One time shot checks could fail because the cluster might be slow - and did not update the service binding at the time of performing the check.

Reply:
Done here b7139c7